examples/with-rosetta/pages/contact.js

```diff
@@ -0,0 +1,31 @@
+import Link from 'next/link'
+import Head from 'next/head'
+import useI18n from '../hooks/use-i18n'
+import { contentLanguageMap } from '../lib/i18n'
+
+const Contact = ({ lng }) => {
+  const i18n = useI18n(lng)
+
+  return (
+    <div>
+      <Head>
+        <meta http-equiv="content-language" content={contentLanguageMap[lng]} />
+      </Head>
+      <h1>{i18n.t('contact.email')}</h1>
+      <div>Current locale: {i18n.activeLocale}</div>
+      <Link href={{ pathname: '/contact', query: { lng: 'de' } }}>
+        <a>Change language SSR to 'de'</a>
+      </Link>
+    </div>
+  )
+}
+
+export async function getServerSideProps({ query }) {
+  return {
+    props: {
+      lng: query.lng || 'en', // OR detect default language based on header OR user setting
+    }, // will be passed to the page component as props
+  }
+}
+
+export default Contact
```

Comment:
Let's move this page to `pages/[lng]/contact.js` and use `getStaticProps`, there's no good use case for having a serverless function decide the language of the page if you can do that at build time :+1:

Reply:
You're right but it must not be a good use case. I want to demonstrate the different modes.

Reply:
We prefer to demonstrate good use cases in examples always over showing usage of some method, we have docs and learning lessons for that.

For this case build time pages is better.

Reply:
Did you get my point? I'd like to demonstrate that the i18n solution works in all modes.

Reply:
Yes, and it surely does, but you don't have to demonstrate it, instead you have to show the community the best way of using it :+1:

Reply:
Would it be worth having 3 separate examples? Whichever you decide should be default behavior can be `with-i18n` and then you can have the two other mode be `with-i18n-(csr|ssg)`. 

While I understand and relate to both points, I do think as an official example, something should be as concise and as close to prod-ready as possible instead of forcing the user to figure out what's actually core/necessary.

> I suggest `with-i18n` since no one will know what "rosetta" is when browsing the directory list ¯\\_(ツ)_/¯

Reply:
@lfades I updated it. @lukeed that was my first idea too but there are other i18n solutions here. I renamed it to `with-rosetta-i18n`

Reply:
Ah, then `with-i18n-rosetta` would be better for grouping 😉

Reply:
Done.